src/api/schema/metrics/filter.rs

```diff
@@ -14,18 +14,39 @@ lazy_static! {
         Arc::new(get_controller().expect("Metrics system not initialized. Please report."));
 }
 
-/// Sums an iteratable of `Metric`, by folding metric values. Convenience function typically
+/// Sums an iteratable of `&Metric`, by folding metric values. Convenience function typically
 /// used to get aggregate metrics.
 fn sum_metrics<'a, I: IntoIterator<Item = &'a Metric>>(metrics: I) -> Option<Metric> {
     let mut iter = metrics.into_iter();
     let m = iter.next()?;
 
     Some(iter.fold(m.clone(), |mut m1, m2| {
-        m1.data.update(&m2.data);
-        m1
+        if m1.data.update(&m2.data) {
+            m1
+        } else {
+            m2.clone()
+        }
     }))
 }
 
+/// Sums an iteratable of `Metric`, by folding metric values. Convenience function typically
+/// used to get aggregate metrics.
+fn sum_metrics_owned<I: IntoIterator<Item = Metric>>(metrics: I) -> Option<Metric> {
```

Comment:
It would be nice to avoid this code duplication and have just one `sum_metrics` function. The two cases that use `sum_metrics_owned` could use the reference version above but at the cost of an unconditional `clone` for each sum produced. Is that an acceptable cost @leebenson?

Reply:
Agreed, I'd support this.